src/utils/log_utils.py

```diff
@@ -7,7 +7,9 @@
 from typing import Optional
 
 
-def init_logging(file_name: Optional[str] = None, level: int = logging.DEBUG) -> None:
+def init_logging(
+    file_name: Optional[str] = None, level: int | str = logging.DEBUG
```

Comment:
This takes an int or string (both valid for log levels) but your validators are only looking for strings.
I don't think we really have any need to set custom logging levels via ints so probably best to have this just typed as str.

Reply:
Actually looking at python's logging it has it's own validation via `_check_level` in it's __init___.py which raises a value error if you give it an invalid string or int so you could also less strictly type your log levels and rely on logging's validation to allow for a bit more customization.

Reply:
Personally I think that having a more clearly defined list of strings (without things like "FATAL") is probably better for the ingestion scripts

Reply:
All fair points. I guess I wanted to treat the interface of init_logging separately from the command-line interface. In the CLI, I think we only want to accept strings because trying to pass in an int (e.g. 20 for INFO) is error-prone and not very readable.

On the other hand, I could maybe imagine a case where someone wants to specify the level in code, and in that case I think we want to encourage them to pass an int like logging.DEBUG, which in this situation will be less error-prone than passing a string. So that was why I wanted to leave both in there.

I also saw that there was some validation in the logging configuration, but I thought it was better for the CLI if the user is getting a nicely formatted Typer error, rather than having to dig through a stack trace to figure out what they did wrong

Reply:
For comparison:

![log_error_1](https://github.com/user-attachments/assets/8773d00e-8f86-4858-86ab-39bfef438c5e)
![log_error_2](https://github.com/user-attachments/assets/b6dcb158-b325-47d0-ae2d-ba1f90eda8c1)